openshift/console
# ODC-6443: Add adv. option to customize route labels

**Fixes**: 
https://issues.redhat.com/browse/ODC-6443

**Analysis / Root cause**: 
As a user, I want have the option to add additional labels to a Route, as I could do in OCP3. See RFE-622.

**Solution Description**: 
Added a new "Labels" field to advanced route option in the add/import flow and the edit page as well.

It is NOT shown for Serverless routes because these are auto-generated and don't accept custom labels. At least exclusive labels for the route.

**Screen shots / Gifs for design review**: 
@openshift/team-devconsole-ux
@serenamarie125 

Default routing options:

![image](https://user-images.githubusercontent.com/139310/146063454-dc85ea01-ecba-4a16-8da8-e6e6cb1b0037.png)

When opening the advanced routing options:

![image](https://user-images.githubusercontent.com/139310/146063545-e3622e92-1550-4155-bcf7-aad634d181a6.png)

**Unit test coverage report**: 
- [x] Added some unit and e2e tests.

**Test setup:**
1. Import from git or container image
    - Open the "advanced routing options"
    - Enter a label
2. Open the Deployment and route detail pages
    - The Deployment must not contain the route-specific labels
    - The Route should contain them :smirk: 
3. Select "Edit your-deployment" and update the labels should also update them.

**Browser conformance**:
